source/libs/get-text-nodes.ts

```diff
@@ -1,12 +1,12 @@
 export default (el: Node): Text[] => {
 	const walker = document.createTreeWalker(el, NodeFilter.SHOW_TEXT);
 	const nodes: Text[] = [];
-	let node: Text;
+	let node;
```

Comment:
This would be more correct:

```suggestion
	let node: Text | null;
```

I try not to use `as` when there are there are other solutions, as `as` can easily hide type errors as it force-casts.

Reply:
I agree and that was my first solution, but this can't be avoided, I think.

`walker.nextNode()` returns a `Node` which doesn't fit `Text | null`

We can change `get-text-nodes` to return `Node[]` (even though we're sure that `createTreeWalker` returns `Text`) but then we'll still have to cast those nodes to `Text` in other places that use APIs not available on `Node`:

`textNode.replaceWith(fragment)` in `show-whitespace` would have to become `(textNode as Text).replaceWith(fragment)`

Reply:
Ugh, I assumed the TS type would be good enough to return the correct type because we specify `SHOW_TEXT`...

Reply:
DOM types aren’t great. Once I suggested a change to add a generic to cloneNode, I think, and they refused because of performance concerns. So… assertions it is